python/ray/tune/trainable/function_trainable.py

```diff
@@ -123,33 +123,35 @@ def create_perm_checkpoint(checkpoint_dir, logdir, step):
         return perm_checkpoint_dir
 
 
+@DeveloperAPI
 class _StatusReporter:
     def __init__(
         self,
-        result_queue,
-        continue_semaphore,
-        end_event,
-        experiment_name=None,
-        trial_name=None,
-        trial_id=None,
-        logdir=None,
-        trial_resources=None,
+        result_queue: queue.Queue,
+        continue_semaphore: threading.Semaphore,
+        end_event: threading.Event,
+        training_iteration_func: Callable[[], int],
+        experiment_name: Optional[str] = None,
+        trial_name: Optional[str] = None,
+        trial_id: Optional[str] = None,
+        logdir: Optional[str] = None,
+        trial_resources: Optional[Union[Resources, PlacementGroupFactory]] = None,
     ):
         self._queue = result_queue
         self._last_report_time = None
         self._continue_semaphore = continue_semaphore
         self._end_event = end_event
+        self._get_training_iteration = training_iteration_func
         self._experiment_name = experiment_name
         self._trial_name = trial_name
         self._trial_id = trial_id
         self._logdir = logdir
         self._last_checkpoint = None
         self._fresh_checkpoint = False
         self._trial_resources = trial_resources
-        # Also used as a marker of whether new `report()` API is being used,
-        # in which case, `_iter` will be incremented from 0 every time `report`
-        # is called.
-        self._iter = None
+        # Mark whether the `session.report()` API is being used,
+        # to throw an error if `tune.report()` is called as well
+        self._has_reported = False
```

Comment:
Is this "has reported using air.session.report"?

If so, can we slightly adjust the name to make this more obvious?

Reply:
Yes, this is for checking air session report usage. Will do!